content/automate/ManageIQ/Infrastructure/VM/Transform/Import.class/__methods__/import_infra_vm.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Description: This method initiates VM import to given infra provider
+#
+
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module Import
+            class ImportInfraVm
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                user_id = @handle.root['user'].id
+
+                options = {
+                  :namespace     => 'Infrastructure/VM/Transform/StateMachines',
+                  :class_name 	 => 'VmImport',
+                  :instance_name => 'default',
+                  :message 	 	   => 'create',
+                  :attrs         => {
+                    'Vm::vm'      => @handle.root['vm'].id,
+                    'name'        => @handle.root['dialog_name'],
+                    'provider_id' => @handle.root['dialog_provider'],
+                    'cluster_id'  => @handle.root['dialog_cluster'],
+                    'storage_id'  => @handle.root['dialog_storage'],
+                    'sparse'      => @handle.root['dialog_sparse'],
+                  },
+                  :user_id       => user_id
+                }
+
+                auto_approve = true
+                @handle.execute('create_automation_request', options, 'admin', auto_approve)
```

Comment:
@mkanoor ah, ok. I thought this has to be `'admin'` when I want to use the auto_approve feature - I understood this is the "approving" user being passed...